dll/win32/shlwapi/propbag.cpp

```diff
@@ -243,12 +243,12 @@ SHCreatePropertyBagOnMemory(_In_ DWORD dwMode, _In_ REFIID riid, _Out_ void **pp
 
     *ppvObj = NULL;
 
-    CMemPropertyBag* pMemBag = new CMemPropertyBag(dwMode);
+    CComPtr<CMemPropertyBag> pMemBag;
+    pMemBag.Attach(new CMemPropertyBag(dwMode));
```

Comment:
```suggestion
    CComPtr<CMemPropertyBag> pMemBag(new CMemPropertyBag(dwMode));
```

Reply:
Needed a fix in `m_cRefs` initialization.

Reply:
If the Attach method had been used, there wouldn't have been the need for fixing the m_cRefs? interesting...

Reply:
`CComPtr` ctor is adding the reference count...
```c
    CComPtr(T *lp)
    {
        p = lp;
        if (p != NULL)
            p->AddRef();
    }
```